### PR TITLE
Making gcc11 the default compiler

### LINKED ISSUE #2705: Make GCC 11 the default
Currently the Debian packages you are providing are coming with gcc/g++ compiler v5.5, which is very outdated. So out-dated I can't even compile some C++20 code, related: #2703. Like std::filesystem.

Anyway, that is why I think MXE should bring ~~GCC9 or GCC10~~ GCC11 as the [default gcc/g++ compiler package](https://pkg.mxe.cc/repos/apt/dists/). So people do not manually need to compile the whole Qt or GTK libraries manually using the [plugins](https://github.com/mxe/mxe/tree/master/plugins/gcc10) on their computer to just (cross-)compile their C++20 project.

Thank you for understanding! 

Regards,
Melroy van den berg

### PR DESCRIPTION
**Pull request by danger89**
- Make **GCC11** the default compiler!
- Move GCC5 to the plugins directory

Fixes #2705

### CODE REVIEW

File: src/binutils.mk [hunk 1]
```diff
@@ -3,8 +3,8 @@
 PKG             := binutils
 $(PKG)_WEBSITE  := https://www.gnu.org/software/binutils/
 $(PKG)_DESCR    := GNU Binutils
-$(PKG)_VERSION  := 2.28
-$(PKG)_CHECKSUM := 6297433ee120b11b4b0a1c8f3512d7d73501753142ab9e2daa13c5a3edd32a72
+$(PKG)_VERSION  := 2.36.1
```
Comment (mabrand) [inline src/binutils.mk:6]: I'd rather see the binutils upgrade in its own commit and PR.

**Comment by mabrand** [comment]
Have you been able to test this, including the new gcc5 overlay? It would be good to know which packages this breaks.

**Comment by danger89** [comment]
I will do some extra testing at the moment. Cross-compile some of those packages takes some time :) But I will cherry pick some packages and see if also the new overlay is working as expected.

**Comment by danger89** [comment]
Also, you are free of course to add additional changes/commits if you want.

**Comment by danger89** [comment]
As I told earlier in the issue, GTK build will fail with gcc11, since it's very outdated. That is why I update the GTK package (out of scope, WIP). But the gcc11 does work!

That being said, this also gave me the ability to build those old `gtk3` and `gtkmm3` packages again using the gcc5 overlay plugin now. And this seems to work fine! So yes, the overlay works 🥳 !

**Comment by danger89** [comment]
TLDR; It's working as expected.

**Comment by danger89** [comment]
Ps. binutils is not upgraded in this PR. I use the current gcc11 dependencies that was defined when using the gcc11 overlay, so the only thing I do I swap the positions of GCC11 ->  src and GCC5 -> plugins.

**Comment by mabrand** [comment]
I realized that just after I wrote the comment, which I deleted not fast enough.

**Review by mabrand — APPROVED** [review]

**Comment by mabrand** [comment]
Let's wait a few days in case anybody else has concerns or ideas.

**Comment by danger89** [comment]
You can also do some testing, but I think you'll come to the same conclusion :)

**Comment by tonytheodore** [comment]
Last time I checked (around late November), gcc10 had these failing packages:
```
windres issues:
    tmp-ffmpeg-i686-w64-mingw32.shared.gcc10/
    tmp-ffmpeg-x86_64-w64-mingw32.shared.gcc10/
hardware resource:
    tmp-pcl-i686-w64-mingw32.shared.gcc10/
    tmp-pcl-i686-w64-mingw32.static.gcc10/
    tmp-pcl-x86_64-w64-mingw32.shared.gcc10/
    tmp-pcl-x86_64-w64-mingw32.static.gcc10/
old 7zip:
    tmp-qtifw-i686-w64-mingw32.static.gcc10/
```
and gcc11 had a longer list that I didn't investigate:
```
    tmp-boost-x86_64-w64-mingw32.shared.gcc11/
    tmp-boost-x86_64-w64-mingw32.static.gcc11/
    tmp-coin-i686-w64-mingw32.shared.gcc11/
    tmp-coin-i686-w64-mingw32.static.gcc11/
    tmp-coin-x86_64-w64-mingw32.shared.gcc11/
    tmp-coin-x86_64-w64-mingw32.static.gcc11/
    tmp-freeimage-i686-w64-mingw32.shared.gcc11/
    tmp-freeimage-i686-w64-mingw32.static.gcc11/
    tmp-freeimage-x86_64-w64-mingw32.shared.gcc11/
    tmp-freeimage-x86_64-w64-mingw32.static.gcc11/
    tmp-gdb-x86_64-w64-mingw32.shared.gcc11/
    tmp-gdb-x86_64-w64-mingw32.static.gcc11/
    tmp-glibmm-i686-w64-mingw32.shared.gcc11/
    tmp-glibmm-i686-w64-mingw32.static.gcc11/
    tmp-glibmm-x86_64-w64-mingw32.shared.gcc11/
    tmp-glibmm-x86_64-w64-mingw32.static.gcc11/
    tmp-ilmbase-i686-w64-mingw32.shared.gcc11/
    tmp-ilmbase-i686-w64-mingw32.static.gcc11/
    tmp-ilmbase-x86_64-w64-mingw32.shared.gcc11/
    tmp-ilmbase-x86_64-w64-mingw32.static.gcc11/
    tmp-intel-tbb-i686-w64-mingw32.shared.gcc11/
    tmp-intel-tbb-i686-w64-mingw32.static.gcc11/
    tmp-intel-tbb-x86_64-w64-mingw32.shared.gcc11/
    tmp-intel-tbb-x86_64-w64-mingw32.static.gcc11/
    tmp-itk-i686-w64-mingw32.shared.gcc11/
    tmp-itk-i686-w64-mingw32.static.gcc11/
    tmp-itk-x86_64-w64-mingw32.shared.gcc11/
    tmp-itk-x86_64-w64-mingw32.static.gcc11/
    tmp-libdnet-i686-w64-mingw32.static.gcc11/
    tmp-libdnet-x86_64-w64-mingw32.static.gcc11/
    tmp-libical-i686-w64-mingw32.shared.gcc11/
    tmp-libical-i686-w64-mingw32.static.gcc11/
    tmp-liblastfm_qt4-i686-w64-mingw32.shared.gcc11/
    tmp-liblastfm_qt4-x86_64-w64-mingw32.shared.gcc11/
    tmp-libtorrent-rasterbar-i686-w64-mingw32.shared.gcc11/
    tmp-libtorrent-rasterbar-i686-w64-mingw32.static.gcc11/
    tmp-llvm-i686-w64-mingw32.shared.gcc11/
    tmp-llvm-i686-w64-mingw32.static.gcc11/
    tmp-llvm-x86_64-w64-mingw32.shared.gcc11/
    tmp-llvm-x86_64-w64-mingw32.static.gcc11/
    tmp-ncurses-i686-w64-mingw32.shared.gcc11/
    tmp-ncurses-i686-w64-mingw32.static.gcc11/
    tmp-ncurses-x86_64-w64-mingw32.shared.gcc11/
    tmp-ncurses-x86_64-w64-mingw32.static.gcc11/
    tmp-qtbase-i686-w64-mingw32.shared.gcc11/
    tmp-qtbase-i686-w64-mingw32.static.gcc11/
    tmp-upx-i686-w64-mingw32.shared.gcc11/
    tmp-upx-i686-w64-mingw32.static.gcc11/
    tmp-upx-x86_64-w64-mingw32.shared.gcc11/
    tmp-upx-x86_64-w64-mingw32.static.gcc11/
    tmp-wt-i686-w64-mingw32.shared.gcc11/
    tmp-wt-i686-w64-mingw32.static.gcc11/
    tmp-xapian-core-i686-w64-mingw32.shared.gcc11/
    tmp-xapian-core-i686-w64-mingw32.static.gcc11/
    tmp-xapian-core-x86_64-w64-mingw32.shared.gcc11/
    tmp-xapian-core-x86_64-w64-mingw32.static.gcc11/
```
I'm happy to switch to gcc11 if:
  - 64bit boost
  - 32bit qt5

are tested and known to work.

**Comment by mabrand** [comment]
Just completed without errors:
`make -j4 JOBS=8 boost~x86_64-w64-mingw32.shared qt5~i686-w64-mingw32.shared`

**Comment by danger89** [comment]
> Last time I checked (around late November), gcc10 had these failing packages

Nice meeting you! Did you ever considered fixing the ci/cd pipeline / netlify? And/or introducing some of those builds in github actions maybe? To avoid regression in a automated way?

In an upcoming pull request I will update whole gtk3 stack: https://github.com/danger89/mxe/pull/1

I was considering adding a test build for gtk in github actions, but I think I should first discuss that with you.

**Comment by danger89** [comment]
> [](https://github.com/danger89/mxe/pull/1)
> 
> I was considering adding a test build for gtk in github actions, but I think I should first discuss that with you.

@tonytheodore Please, how can I discuss this with you? How do you want me to contact you?
